sdk/identity/Azure.Identity/tests/VisualStudioCredentialLiveTests.cs

```diff
@@ -0,0 +1,186 @@
+            };
+
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+            Assert.CatchAsync<OperationCanceledException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), cts.Token));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_CanceledOnProcessRun()
+        {
+            var cts = new CancellationTokenSource();
+            var fileSystem = CreateTestFileSystem();;
+            var testProcess = new TestProcess { Timeout = 10000 };
+            testProcess.Started += (o, e) => cts.Cancel();
+
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+            Assert.CatchAsync<OperationCanceledException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), cts.Token));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_NoVsInstalled()
+        {
+            var fileSystem = CreateTestFileSystem();
+            fileSystem.FileExistsHandler = p => false;
+
+            var (_, _, processOutput) = CreateTestToken();
+            var testProcess = new TestProcess { Output = processOutput };
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+            Assert.ThrowsAsync<CredentialUnavailableException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), CancellationToken.None));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_NoJsonFileFound()
+        {
+            var (_, _, processOutput) = CreateTestToken();
+            var testProcess = new TestProcess { Output = processOutput };
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, new TestFileSystemService(), new TestProcessService(testProcess)));
+            Assert.ThrowsAsync<CredentialUnavailableException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), CancellationToken.None));
+        }
+
+        [Test]
+        public void AuthenticateWithVsCredential_BrokenJsonFileFound()
+        {
+            var (_, _, processOutput) = CreateTestToken();
+            var testProcess = new TestProcess { Output = processOutput };
+            var fileSystem = new TestFileSystemService { ReadAllHandler = p => "{\"Some\": false}" };
+            var credential = InstrumentClient(new VisualStudioCredential(default, default, fileSystem, new TestProcessService(testProcess)));
+            Assert.ThrowsAsync<AuthenticationFailedException>(async () => await credential.GetTokenAsync(new TokenRequestContext(new[]{"https://vault.azure.net/"}), CancellationToken.None));
```

Comment:
Should it be CredentialUnavailableException here?

Reply:
@schaabs , should we always throw `CredentialUnavailableException`?

Reply:
No, only when authenticating through VS "unavailble". In other words VS is not installed, or no user is logged into VS. In the case that authentication failed due to an unexpected error or exception this should throw `AuthenticationFailedException`